shell/key-bindings.fish

```diff
@@ -11,6 +11,9 @@
 # - $FZF_ALT_C_COMMAND
 # - $FZF_ALT_C_OPTS
 
+status is-interactive; or return 0
```

Comment:
Let's see if this works.

```suggestion
status is-interactive; or exit
```

Reply:
@junegunn, I don't think this should be done, as argued in 5cb5bbdc32fa581c5e2c93552cf60a9ded4d147a.

Using `exit` defeats nearly the whole idea of the check, as it kills of any non-interactive script, that would source our script by accident.

Reply:
Welcome to the wonderful land of fish (which I don't fathom).

```sh
$ echo 'echo before-exit; exit; echo after-exit' > /tmp/x
$ fish -c 'echo before-source; source /tmp/x; echo after-source'

  before-source
  before-exit
  after-source

$ sh -c 'echo before-source; source /tmp/x; echo after-source'

  before-source
  before-exit
```

Reply:
Reference: https://github.com/NixOS/nix/pull/7105

Reply:
But I don't think this is really needed anyway. Because the file only defines a function and does not alter the state of the current shell until the user manually calls the function afterward. So I suspect not many fish users will use it in non-autoload setups.

Reply:
> Welcome to the wonderful land of fish (which I don't really understand either).

Hmm... indeed... seems to work. Weird.

From the comment in https://github.com/fish-shell/fish-shell/commit/3359e5d2e9bcbf19d1652636c8e448a6889302ae I assumed that `exit` would cause the fish shell to exit, bu looking at [fish-`exit`’s documentation](https://fishshell.com/docs/2.5/commands.html#exit) it’s even documented not to do so:
> exit causes fish to exit. If STATUS is supplied, it will be converted to an integer and used as the exit code. Otherwise, the exit code will be that of the last command executed.
> If exit is called while sourcing a file (using the source builtin) the rest of the file will be skipped, but the shell itself will not exit.

I would however suggest that we use `exit 0`... otherwise we get a exit status of `1` in the real code.

Reply:
Force pushed a commit that uses `exit 0` and has updated commit message.

Reply:
But in #3467.